src/main/kotlin/org/xodium/vanillaplus/handlers/REINetworkHandler.kt

```
@@ -0,0 +1,131 @@
+import java.io.ByteArrayOutputStream
+import java.io.DataInputStream
+import java.io.DataOutputStream
+import java.util.*
+
+/** Handles REI network communication between the server and clients. */
+internal object REINetworkHandler : PluginMessageListener {
+    private const val HANDSHAKE_PACKET_ID = 0
+    private const val RECIPE_TRANSFER_PACKET_ID = 1
+    private const val LATEST_PROTOCOL_VERSION = 19
+
+    private val playerProtocolVersions = mutableMapOf<UUID, Int>()
```

detekt (1.23.8)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)